cpp/include/cudf/detail/reduction_functions.hpp

```diff
@@ -35,13 +35,15 @@ namespace reduction {
  *
  * @param col input column to compute sum
  * @param output_dtype data type of return type and typecast elements of input column
- * @param stream CUDA stream used for device memory operations and kernel launches.
+ * @param init initial value of the sum
+ * @param stream CUDA stream used for device memory operations and kernel launches
  * @param mr Device memory resource used to allocate the returned scalar's device memory
- * @return Sum as scalar of type `output_dtype`.
+ * @return Sum as scalar of type `output_dtype`
  */
 std::unique_ptr<scalar> sum(
   column_view const& col,
   data_type const output_dtype,
+  std::optional<std::reference_wrapper<const scalar>> init,
```

Comment:
Need to fix style here: `scalar const`. This also applies to all similar places below.